src/main/java/org/spongepowered/api/asset/Asset.java

```diff
@@ -77,6 +77,33 @@ default void copyToFile(Path output) throws IOException {
     }
 
     /**
+     * Copies this Asset to the specified 'outputDirectory' {@Link Path}.
+     *
+     * @param outputDirectory The directory to copy to
+     * @throws IOException
+     */
+    default void copyToDirectory(Path outputDirectory) throws IOException {
+        checkNotNull(outputDirectory, "outputDirectory");
+        copyToFile(outputDirectory.resolve(this.getFileName()));
```

Comment:
Should an asset have a concept of a sub-directory? or should that be delegated to plugins to use if they need anything further?

In my mind copyToDirectory would be used to copy default configs + examples to the correct directories inside the plugin folders.

But if they need sub-directories that falls apart somewhat, as assets currently only know about their own filenames, and not whether they are in a sub-directory. So plugins would be responsible for discovering their assets, and setting the appropriate sub-directory. I guess that isn't so bad, but would appreciate any input others might have.